### PR TITLE
Bundle plugins

### LINKED ISSUE #8: Consider bundling plugins
For https://github.com/calebeby/eslint-config, I discovered an approach for removing the need to [install all of the necessary plugins on each project](https://github.com/cloudfour/eslint-config#installation) by `require`ing the plugins' rules from the ESLint config, moving them under a namespace, and re-exporting them, forming a sort of super-plugin that exports all the rules from the other plugins, and a recommended config.

https://github.com/calebeby/eslint-config/blob/master/src/rules.js

I could use a similar approach here, which would make installing this ESLint config to other projects much easier. Also, this means that updating sub-plugins versions would mean updating them in this project, instead of trying to keep all the versions in sync across all Cloud Four projects.

@gerardo-rodriguez @Josh68

### PR DESCRIPTION
**Pull request by calebeby**
Closes #8 

You can try this out locally with:

`npm i`
`npm link && npm link @cloudfour/eslint-plugin` this links itself as one of its own dependencies
`npm run build` this creates a snapshot of the merged ESLint config into `dist/config.js`. You can look through the generated config and see if it is reasonable. All stylistic rules that Prettier deals with are removed.

You can try adding this to another project by referencing it like this: https://github.com/cloudfour/eslint-config/compare/feature/bundle-plugins?expand=1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R59, and running `npm link @cloudfour/eslint-plugin` in that directory.

### CODE REVIEW

**Comment by calebeby** [comment]
Question: do we want to default to using `babel-eslint`? I think it might make sense, so that it allows us to use experimental syntax like rest spread and class properties

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/dist
```
Comment (gerardo-rodriguez) [inline .eslintignore]: Missing newline at end of file?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 node_modules
+/dist
```
Comment (gerardo-rodriguez) [inline .gitignore]: Missing new line at end of file.

File: .npmrc [hunk 3]
```diff
@@ -1 +1,3 @@
-tag-version-prefix = ""
+# We want npm to not create the tag when the version is updated (in PRs)
+# instead we want the tag created after the PR is merged, on master, on publish
+git-tag-version=false
```
Comment (gerardo-rodriguez) [inline .npmrc]: Missing new line at end of file.

File: .prettierignore [hunk 4]
```diff
@@ -0,0 +1 @@
+/dist
```
Comment (gerardo-rodriguez) [inline .prettierignore]: Missing new line at end of file.

**Comment by gerardo-rodriguez** [comment]
> Question: do we want to default to using `babel-eslint`? I think it might make sense, so that it allows us to use experimental syntax like rest spread and class properties. The default ESLint parser will work most of the time, except when we need experimental syntax.

@calebeby Good question. Seems like we should? Are there any drawbacks to doing so?

**Comment by calebeby** [comment]
It will be slower, because it has to parse it as a Babel AST and then traverse the tree to convert it to be espree-compatible, but it probably won't be noticeably slower.

**Comment by gerardo-rodriguez** [comment]
> It will be slower, because it has to parse it as a Babel AST and then traverse the tree to convert it to be espree-compatible, but it probably won't be noticeably slower.

Seems fine to me.

**Comment by gerardo-rodriguez** [comment]
@calebeby When I tried to run `npm run build`, I got this error:

```
$ npm run build

> @cloudfour/eslint-plugin@1.0.0 build /Users/Rodriguez/Development/projects/cloudfour/repos/eslint-config
> node build.js

fs.js:646
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: ENOENT: no such file or directory, open 'dist/config.js'
    at Object.fs.openSync (fs.js:646:18)
    at fs.writeFileSync (fs.js:1299:33)
    at Object.<anonymous> (/Users/Rodriguez/Development/projects/cloudfour/repos/eslint-config/build.js:36:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @cloudfour/eslint-plugin@1.0.0 build: `node build.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @cloudfour/eslint-plugin@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Rodriguez/.npm/_logs/2018-11-15T18_56_28_052Z-debug.log
```
🤔

**Comment by calebeby** [comment]
@gerardo-rodriguez I think it should work now

**Comment by gerardo-rodriguez** [comment]
> @gerardo-rodriguez I think it should work now

Yep! `npm run build` worked this time. 

Does the README need to be updated due to these changes?

**Comment by gerardo-rodriguez** [comment]
> Fix `extends` examples in README

Great catch! 😬

**Comment by calebeby** [comment]
@gerardo-rodriguez it looks like the default ESLint parser does support object spread, as long as `ecmaVersion` is set to `2018`. I think I'm not gonna switch to `babel-eslint` because class properties are the only experimental feature that I could see us using, and I don't think we are using that at all. For projects that need it, we can override the parser

**Pull request merged**